pkg/model/iam/iam_builder.go

```diff
@@ -244,6 +234,41 @@ func (b *IAMPolicyBuilder) BuildAWSIAMPolicy() (*IAMPolicy, error) {
 	return p, nil
 }
 
+// addEC2Permissions updates the IAM Policy with statements granting tailored
+// access to EC2 resources, depending on the instance role
+func addEC2Permissions(p *IAMPolicy, iamPrefix string, b *IAMPolicyBuilder, wildcard stringorslice.StringOrSlice) {
+	if (b.Role == api.InstanceGroupRoleNode) || (b.Role == api.InstanceGroupRoleMaster) {
+		p.Statement = append(p.Statement, &IAMStatement{
+			Effect:   IAMStatementEffectAllow,
+			Action:   stringorslice.Slice([]string{"ec2:Describe*"}),
+			Resource: wildcard,
+		})
+	}
+
+	if b.Role == api.InstanceGroupRoleMaster {
+		p.Statement = append(p.Statement,
+			&IAMStatement{
+				Effect: IAMStatementEffectAllow,
+				Action: stringorslice.Slice([]string{
+					"ec2:ModifyInstanceAttribute",
+					"ec2:CreateRoute",
```

Comment:
I am worried that there might be more permissions in this list, but I don't know what they would be so...